src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -133,25 +133,136 @@ CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_ConfigureChipStack()
 template <class ImplClass>
 CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_GetVendorName(char * buf, size_t bufSize)
 {
-    ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_DEVICE_VENDOR_NAME), CHIP_ERROR_BUFFER_TOO_SMALL);
-    strcpy(buf, CHIP_DEVICE_CONFIG_DEVICE_VENDOR_NAME);
-    return CHIP_NO_ERROR;
+    CHIP_ERROR err;
+    size_t vlen;
+
+    err = Impl()->ReadConfigValueStr(ImplClass::kConfigKey_VendorName, buf, bufSize, vlen);
+    if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
+    {
+        ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_DEVICE_VENDOR_NAME), CHIP_ERROR_BUFFER_TOO_SMALL);
+        strcpy(buf, CHIP_DEVICE_CONFIG_DEVICE_VENDOR_NAME);
+    }
```

Comment:
This assumes null-terminated storage. If storage is corrupted or tempered, it's possible the underlying backing will not be null-terminated. Please change *every* string accessor to not assume *stored* null-termination, and instead zero-out the output buffer and ensure it is always null-terminated on exit.

Also, there is a lot of repetition of boilerplate, which could easily be replaced by a utility method in the anonymous namespace e.g. `LoadStringKey(key, max_size output_str, output_str_size)`